icontrol/session.py

```diff
@@ -292,8 +311,11 @@ def __init__(self, username, password, **kwargs):
             self.session.auth = (username, password)
 
         # Set state as indicated by ancestral code.
-        self.session.verify = False  # XXXmake TOFU
-        self.session.headers.update({'Content-Type': 'application/json'})
+        self.session.verify = False  # XXX make TOFU
+        self.session.headers.update({
```

Comment:
Ok, so I was working on fix for this for you guys, I should have assigned it to myself. This probably isn't the right way to do this because you are overwriting every thing that was there before which may not be what you want to do.

I was working on a bottom up approach from iControl Rest that preserved any User-Agent headers that were there and appended an additional agent string to it for the libraries.  I was planning to do the same here.

What I did was add a method to the iControl library that allowed you to append a user agent and then called that from iControl __init__ and then I figured SDK could also call it to append it's user agent and then any users of the SDK could also call it to append theirs like openstack-agent.